controllers/metal3machine_controller.go

```diff
@@ -248,6 +270,37 @@ func (r *Metal3MachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+// ClusterToMetal3Machines is a handler.ToRequestsFunc to be used to enqeue
+// requests for reconciliation of Metal3Machines.
+func (r *Metal3MachineReconciler) ClusterToMetal3Machines(o handler.MapObject) []ctrl.Request {
+	result := []ctrl.Request{}
+	c, ok := o.Object.(*capi.Cluster)
+
+	if !ok {
+		r.Log.Error(errors.Errorf("expected a Cluster but got a %T", o.Object), "failed to get Metal3Machine for Cluster")
+		return nil
+	}
+
+	labels := map[string]string{capi.ClusterLabelName: c.Name}
+	capiMachineList := &capi.MachineList{}
+	if err := r.Client.List(context.TODO(), capiMachineList, client.InNamespace(c.Namespace), client.MatchingLabels(labels)); err != nil {
+		r.Log.Error(err, "failed to list Metal3Machines")
+		return nil
+	}
+	for _, m := range capiMachineList.Items {
+		if m.Spec.InfrastructureRef.Name == "" {
+			continue
+		}
+		name := client.ObjectKey{Namespace: m.Namespace, Name: m.Name}
```

Comment:
It should not be m.Name, it should be m.Spec.InfrastructureRef.Name

Reply:
Done